DBTools/python/getSiblings.py

```diff
@@ -0,0 +1,331 @@
+
+    #get list of files from DBS
+    def getFileList(self, dataset, api, debug=False):
+        files = api.listFiles (dataset = dataset, detail=True)
+        names = [x['logical_file_name'] for x in files if x['is_file_valid']]
+        if debug: return names[:10] #for debugging
+        else: return names
+
+    #get list of local files
+    def getLocalFileList(self, dataDir):
+        fileList = []
+        for filename in os.listdir(dataDir):
+            if not filename.endswith('root'): continue
+            fileList.append(dataDir + filename)
+        return fileList
+
+    #match files using lumi blocks
+    def getLumiMatching(self, output_json, prod, inputJSON=None, sibJSON=None):
+        if not inputJSON:
+            if prod != 'local' and prod != 'allLocal': 
+                self.getLumiBlocksDBS(self.inputFileList, self.dbsapi_in, 'input') #special case when forcing lumi matching on non local
+            else: 
+                print("getting lumi blocks for input files")
+                self.getLumiBlocks(self.inputFileList, 'input')
+            input_fin = open('input.json', 'r')
+        else:
+            input_fin = open(inputJSON, 'r')
+
+        if not sibJSON:
+            if prod != 'allLocal': 
+                self.getLumiBlocksDBS(self.siblingFileList, self.dbsapi_out, 'sibling') #special case when dataset is not local
+            else: 
+                self.getLumiBlocks(self.siblingFileList, 'sibling')
+            sib_fin = open('sibling.json', 'r')
+        else:
+            sib_fin = open(sibJSON, 'r')
+
+        inputLumiBlocks = json.load(input_fin)
+        siblingLumiBlocks = json.load(sib_fin)
+
+        dict_out = {}
+
+        for inputFile in inputLumiBlocks.keys():
+            print("Looking for lumis in {}".format(inputFile))
+            lumis_in = np.array(inputLumiBlocks[inputFile])
+            for sibFile in siblingLumiBlocks.keys():
+                lumis_sib = np.array(siblingLumiBlocks[sibFile])
+                if np.intersect1d(lumis_in, lumis_sib).size > 0: 
+                    if inputFile in dict_out.keys():
+                        dict_out[inputFile].append(sibFile)
+                    else:
+                        dict_out[inputFile] = [sibFile]
+            if not inputFile in dict_out:
+                print("File {0} has no matching run/lumi in the sibling dataset".format(inputFile))
+
+        f_out = open(output_json, 'w')
+        json_dict = json.dumps(dict_out)
+        f_out.write(json_dict)
+        f_out.close()
+
+    def getFilesFromList(self, jobNumber, nJobs):
+        sys.path.append(os.getcwd())
+
+        Label = args.inputFiles
+        datasetInfo = importlib.import_module('datasetInfo_' + Label +'_cfg', package=None)
+
+        #exec("import datasetInfo_" + Label +"_cfg as datasetInfo")
+
+        filesPerJob = int(1) #int (math.floor (len (datasetInfo.listOfFiles) / nJobs))
+        residualLength = int(10) #int(len(datasetInfo.listOfFiles)%nJobs)
```

Comment:
Is line 282 really needed?

Is there a reason to keep comments in lines 284 and 285?

Reply:
these were supposed to be changed back, fixed now